Code review pull request

Mboesn opened the pull request:
This pull request has all the code being merged into an empty branch. 

This is done in order to have an easy way to review the entire code and write comments about everything.

Mboesn committed:
added all the code back in order to code review

lycaon7 commented on src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:
```diff
@@ -0,0 +1,60 @@
+    public double[] getValues() {
+        return new double[]{masterMotor.getSelectedSensorPosition()};
+    }
+}
```
Add `configureLogName`

lycaon7 commented on src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:
```diff
@@ -0,0 +1,60 @@
+public class ShooterSS extends SubsystemBase implements TestableSubsystem, Loggable {
+    private final ShooterConstants constants;
+    private final TrigonTalonFX masterMotor;
+    public int areaCounter;
```
Why is this here?

lycaon7 commented on src/main/java/frc/robot/subsystems/spinner/SpinnerCMD.java:
```diff
@@ -0,0 +1,65 @@
+
+    @Override
+    public void execute() {
+        // Checks if motor is stalling as a result of a ball stuck in the intake and
```
ball stuck in the Spinner

lycaon7 commented on src/main/java/frc/robot/subsystems/spinner/SpinnerSS.java:
```diff
@@ -0,0 +1,96 @@
+
+    /*
+    <NOTE>
+    The code for the "Colour Spinney Thing" - Krombein,N(2021) is not relevant for the current challenges therefore have been commented out.
```
Great meme

lycaon7 commented on src/main/java/frc/robot/utilities/OIMap.java:
```diff
@@ -0,0 +1,33 @@
+            return axis;
+        }
+    }
+}
```
new line at end of file

lycaon7 commented on src/main/java/frc/robot/utilities/TrigonPlaystationController.java:
```diff
@@ -0,0 +1,193 @@
+            rumbleAmount--;
+        }
+    }
+}
```
new line at end of file

Mboesn commented on src/main/java/frc/robot/subsystems/shooter/ShooterCMD.java:
```diff
@@ -0,0 +1,193 @@
+        double output;
+        TBHController.setSetpoint(desiredVelocity);
+        PIDController.setSetpoint(desiredVelocity);
+        if (ledSS != null)
```
remove

Mboesn commented on src/main/java/frc/robot/subsystems/shooter/ShooterCMD.java:
```diff
@@ -0,0 +1,193 @@
+    @Log(name = "Shooter/Desired Velocity")
+    private DoubleSupplier desiredVelocitySupplier;
+    private double desiredVelocity;
+    private double f;
```
change name

Mboesn commented on src/main/java/frc/robot/subsystems/shooter/ShooterCMD.java:
```diff
@@ -0,0 +1,193 @@
+            if (limelight.hasTarget())
+                Shoot();
+            else {
+                if (ledSS != null)
```
remove

Mboesn commented on src/main/java/frc/robot/subsystems/shooter/ShooterSS.java:
```diff
@@ -0,0 +1,60 @@
+    @Log(name = "Shooter/Velocity")
+    public double getVelocityRPM() {
+        // timed by 600 to convert to minutes and divide by 2048 to convert to revolutions
+        return masterMotor.getSelectedSensorVelocity() * 600 / 2048;
```
maguc nums

Mboesn commented on src/main/java/frc/robot/subsystems/spinner/SpinnerSS.java:
```diff
@@ -0,0 +1,96 @@
+
+    @Log(name = "Stator current")
+    public double getStatorCurrent() {
+        return motor.getStatorCurrent();
```
absolute

Mboesn commented on src/main/java/frc/robot/subsystems/spinner/SpinnerSS.java:
```diff
@@ -0,0 +1,96 @@
+
+    @Log(name = "Is stalled")
+    public boolean isStalled() {
+        return motor.getStatorCurrent() > constants.STALL_CURRENT_LIMIT;
```
absolute

Mboesn commented on src/main/java/frc/robot/utilities/FeedforwardConstants.java:
```diff
@@ -0,0 +1,34 @@
+package frc.robot.utilities;
+
+public class FeedforwardConstants {
+    public final double mCoef;
```
private with getters and setters

Mboesn commented on src/main/java/frc/robot/utilities/PIDFCoefs.java:
```diff
@@ -0,0 +1,182 @@
+package frc.robot.utilities;
```
package you dumb fuck

Mboesn commented on src/main/java/frc/robot/utilities/SwerveConstants.java:
```diff
@@ -0,0 +1,42 @@
+
+public class SwerveConstants {
+
+    public TrigonTalonFX speedMotor;
```
private with getters and setters

Mboesn commented on src/main/java/frc/robot/utilities/TrigonPIDController.java:
```diff
@@ -0,0 +1,66 @@
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+
+public class TrigonPIDController extends PIDController {
+    private double f;
```
kF

Mboesn commented on src/main/java/frc/robot/utilities/TrigonPlaystationController.java:
```diff
@@ -0,0 +1,193 @@
+    private int rumbleAmount;
+    private Notifier notifier;
+
+    public TrigonPlaystationController(int port) {
```
add option to get controller for easy access to obscure functions

Mboesn commented on src/main/java/frc/robot/utilities/TrigonProfiledPIDController.java:
```diff
@@ -0,0 +1,66 @@
+import edu.wpi.first.wpilibj.smartdashboard.SendableBuilder;
+
+public class TrigonProfiledPIDController extends ProfiledPIDController {
+    private double f;
```
kF

Mboesn commented on src/main/java/frc/robot/vision/LedMode.java:
```diff
@@ -0,0 +1,15 @@
+    public int getValue() {
+      return value;
+    }
+}
```
add option to check get controller itself for easy access to obscure functions

Mboesn commented on src/main/java/frc/robot/vision/limelights/PitcherLimelight.java:
```diff
@@ -0,0 +1,102 @@
+ * Class for Limelight when it's on the pitcher. It gets two sets of constants,
+ * one for when the hood is extended and one for when it is retracted.
+ */
+public class PitcherLimelight extends VanillaLimelight {
```
remove pitcher limelihgt, add it to pitcherSS

Mboesn commented on src/main/java/frc/robot/vision/limelights/PitcherLimelight.java:
```diff
@@ -0,0 +1,102 @@
+    /**
+     * @return the vector between the middle of the robot and the target.
+     */
+    private Vector2d calculateVector() {
```
this isnt used, does it need to exist?

Mboesn commented on src/main/java/frc/robot/vision/limelights/PitcherLimelight.java:
```diff
@@ -0,0 +1,102 @@
+    public double calculateDistanceFromTower() {
+        double y = getTy();
+        if (hoodExtended())
+            return 4.15*Math.pow(Math.E, -0.0604*y) + 0.8;
+        else
+            return 4.15*Math.pow(Math.E, -0.0604*y) + 0.8;
+    }
```
magic numbers

Mboesn commented on src/main/java/frc/robot/vision/limelights/VanillaLimelight.java:
```diff
@@ -0,0 +1,220 @@
+    /**
+     * @param tableKey the key of the limelight - if it was changed.
+     */
+    public VanillaLimelight(String tableKey, LimelightConstants constants) {
```
change name to limelight

Mboesn commented on src/main/java/frc/robot/subsystems/intake/IntakeCMD.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Override
+    public void end(boolean interrupted) {
+        if(ledSS != null)
```
remove

Mboesn commented on src/main/java/frc/robot/commands/SensorTestCMD.java:
```diff
@@ -0,0 +1,73 @@
+
+import java.util.HashMap;
+
+public class SensorTestCMD extends CommandBase {
```
test this class **improtant**

Mboesn commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+    private final DashboardController dashboardController;
+    private final PitcherLimelight limelight;
+
+    private ShooterCMD shooterCMD;
+    private CalibrateShooterKfCMD calibrateShooterKfCMD;
+    private GenericCalibrateKF calibrateLoaderKfCMD;
+    private SupplierFieldDriveCMD supplierFieldDriveCMD;
+    private TurnToTargetCMD turnToTargetCMD;
+    private TurnAndPositionToTargetCMD turnAndPositionToTargetCMD;
+    private ToggleMotorsModeCMD toggleMotorsModeCMD;
+
+    private TrigonSwerveControllerCMDGP motionTest;
+    private IntakeOpenerCMD intakeCMD;
+
+    private ShootCMDGP shootCMDGP;
+    private ShootWithPitcherCMDGP ShootWithPitcherCMDGP;
+    private CollectCMDGP collectCMDGP;
+
+    private RunWhenDisabledCommand resetDirection;
```
nope they cant

Mboesn commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     * Initializes all commands.
+     */
+    public void initializeCommands() {
+        SmartDashboard.putNumber("Shooter/Desired Velocity", 0);
```
no

Mboesn reviewed (commented):
Changes in main code

Mboesn commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        driverXboxController.getButtonB().whileHeld(
+                new ShootCMDGP(subsystemContainer, robotConstants, limelight).withInterrupt(this::cancelShooterCMD));
+
+        // robot.win=true
```
yes

Mboesn commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        driverXboxController.getButtonB().whileHeld(
+                new ShootCMDGP(subsystemContainer, robotConstants, limelight).withInterrupt(this::cancelShooterCMD));
+
+        // robot.win=true
```
never

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+    private TrigonSwerveControllerCMDGP motionTest;
+    private IntakeOpenerCMD intakeCMD;
+
+    private ShootCMDGP shootCMDGP;
+    private ShootWithPitcherCMDGP ShootWithPitcherCMDGP;
+    private CollectCMDGP collectCMDGP;
```
`CMDGP` to `CG`

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     */
+    public RobotContainer() {
+
+        Logger.configureLoggingAndConfig(this, true);
```
explain in a comment

Mboesn commented on src/main/java/frc/robot/commands/CommandWithoutSubsystemRequirement.java:
```diff
@@ -0,0 +1,32 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+/**
+ * Commands without subsystem requirement should be in this folder.
+ * For example, this command.
+ */
+public class CommandWithoutSubsystemRequirement extends CommandBase {
+  /**
+   * Creates a new CommandWithoutSubsystemRequirement.
+   */
+  public CommandWithoutSubsystemRequirement() {
+  }
+
+  @Override
+  public void initialize() {
+  }
+
+  @Override
+  public void execute() {
+  }
+
+  @Override
+  public void end(boolean interrupted) {
+  }
+
+  @Override
+  public boolean isFinished() {
+    return false;
+  }
+}
```
cool

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        bindOverrideCommands();
+        setShuffleBoard();
+
+        SmartDashboard.putData("Shooter Command", shooterCMD);
+        SmartDashboard.putData("CalibrateShooterKfCMD", calibrateShooterKfCMD);
+        SmartDashboard.putData("CalibrateLoaderKfCMD", calibrateLoaderKfCMD);
+        SmartDashboard.putData("TurnToTargetCMD", turnToTargetCMD);
+        SmartDashboard.putData("TurnAndPositionToTargetCMD", turnAndPositionToTargetCMD);
+        SmartDashboard.putData("TrigonSwerveControllerCMDGP", motionTest);
```
Name them consistantly

Mboesn commented on src/main/java/frc/robot/commands/GenericCalibrateKF.java:
```diff
@@ -0,0 +1,90 @@
+import frc.robot.utilities.Logger;
+
+public class GenericCalibrateKF extends CommandBase {
+    private final static int PIDF_MAX_OUTPUT = 1023;
+    private final static double CODE_ITERATION_RATE = 0.02;
```
these are class constants, they should be used more often

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     */
+    public void initializeCommands() {
+        SmartDashboard.putNumber("Shooter/Desired Velocity", 0);
+        shooterCMD = new ShooterCMD(subsystemContainer.SHOOTER_SS, null, robotConstants.shooterConstants,
+                () -> SmartDashboard.getNumber("Shooter/Desired Velocity", 0));
+        calibrateShooterKfCMD = new CalibrateShooterKfCMD(subsystemContainer.SHOOTER_SS,
```
explain in a comment

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     * Initializes all commands.
+     */
+    public void initializeCommands() {
+        SmartDashboard.putNumber("Shooter/Desired Velocity", 0);
```
It's not a command that you're initializing, so don't put it here

Mboesn replied:
this was required because that value is given to the command in the constructor and if you attempt to get a value without sending it first weird and annoying things happen

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+                () -> Math.signum(driverXboxController.getX(Hand.kRight)) * Math.pow(driverXboxController.getX(Hand.kRight), 2) / 1.5,
+                () -> Math.signum(driverXboxController.getY(Hand.kRight)) * Math.pow(driverXboxController.getY(Hand.kRight), 2) / 1.5,
+                () -> Math.signum(driverXboxController.getX(Hand.kLeft)) * Math.pow(driverXboxController.getX(Hand.kLeft), 2) / 1.5);
+
+        motionTest = new TrigonSwerveControllerCMDGP(subsystemContainer.DRIVETRAIN_SS,
+                robotConstants.motionProfilingConstants, AutoPath.Test);
+
+        calibrateShooterKfCMD = new CalibrateShooterKfCMD(subsystemContainer.SHOOTER_SS,
+                robotConstants.shooterConstants);
+        calibrateLoaderKfCMD = new GenericCalibrateKF(subsystemContainer.LOADER_SS,
+                robotConstants.loaderConstants.FEEDFORWARD_CONSTANTS);
```
explain them in comments

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        calibrateLoaderKfCMD = new GenericCalibrateKF(subsystemContainer.LOADER_SS,
+                robotConstants.loaderConstants.FEEDFORWARD_CONSTANTS);
+
+        shootCMDGP = new ShootCMDGP(subsystemContainer, robotConstants, limelight);
+        ShootWithPitcherCMDGP = new ShootWithPitcherCMDGP(subsystemContainer, robotConstants, limelight);
```
explain in comments, and rename cmdgp to cg

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        ShootWithPitcherCMDGP = new ShootWithPitcherCMDGP(subsystemContainer, robotConstants, limelight);
+        collectCMDGP = new CollectCMDGP(subsystemContainer, robotConstants);
+        intakeCMD = new IntakeOpenerCMD(true, subsystemContainer.INTAKE_OPENER_SS, robotConstants.intakeOpenerConstants);
+        turnToTargetCMD = new TurnToTargetCMD(subsystemContainer.DRIVETRAIN_SS, limelight,
+                robotConstants.visionConstants, Target.PowerPort);
+        turnAndPositionToTargetCMD = new TurnAndPositionToTargetCMD(subsystemContainer.DRIVETRAIN_SS, limelight,
+                robotConstants.visionConstants, Target.PowerPort);
+        toggleMotorsModeCMD = new ToggleMotorsModeCMD(subsystemContainer.DRIVETRAIN_SS);
+        resetDirection = new RunWhenDisabledCommand(() -> {
+            subsystemContainer.DRIVETRAIN_SS.resetGyro();
+            subsystemContainer.DRIVETRAIN_SS.resetOdometry(new Pose2d());
+        });
+        resetDirection.addRequirements(subsystemContainer.DRIVETRAIN_SS);
```
explain in comments and rename gmdgp to cg

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        ShootWithPitcherCMDGP = new ShootWithPitcherCMDGP(subsystemContainer, robotConstants, limelight);
+        collectCMDGP = new CollectCMDGP(subsystemContainer, robotConstants);
+        intakeCMD = new IntakeOpenerCMD(true, subsystemContainer.INTAKE_OPENER_SS, robotConstants.intakeOpenerConstants);
+        turnToTargetCMD = new TurnToTargetCMD(subsystemContainer.DRIVETRAIN_SS, limelight,
+                robotConstants.visionConstants, Target.PowerPort);
+        turnAndPositionToTargetCMD = new TurnAndPositionToTargetCMD(subsystemContainer.DRIVETRAIN_SS, limelight,
```
It's not turn to target, it's turn to port

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     * the commands.
+     */
+    public void bindDriverCommands() {
+        driverXboxController.getRightBumper().whenHeld(collectCMDGP).whenReleased(intakeCMD);
+        driverXboxController.getButtonY().whenPressed(resetDirection);
+        InstantCommand changeDriveRotation = new InstantCommand(() -> {
+            subsystemContainer.DRIVETRAIN_SS.setAngle(subsystemContainer.DRIVETRAIN_SS.getAngle() + 90);
+            subsystemContainer.DRIVETRAIN_SS.resetOdometry(new Pose2d(0, 0, Rotation2d.fromDegrees(subsystemContainer.DRIVETRAIN_SS.getAngle())));
+        });
+        changeDriveRotation.addRequirements(subsystemContainer.DRIVETRAIN_SS);
+        driverXboxController.getLeftBumper().whenPressed(changeDriveRotation);
+        driverXboxController.getButtonX().whenPressed(
+                new InstantCommand(subsystemContainer.PITCHER_SS::toggleSolenoid, subsystemContainer.PITCHER_SS));
+        driverXboxController.getButtonB().whileHeld(
```
explain in comments

Mboesn commented on src/main/java/frc/robot/commands/MoveMovableSubsystem.java:
```diff
@@ -0,0 +1,47 @@
+     * This command can move any subsystem that implements MovableSubsystem and runs
+     * forever.
+     */
+    public MoveMovableSubsystem(MovableSubsystem subsystem, DoubleSupplier power) {
+        this(subsystem, power, () -> false);
+    }
+
+    /**
+     * This command can move any subsystem that implements MovableSubsystem, it uses
+     * isFinished supplier is checked in the isFinished method for checking if the
+     * command is done.
+     */
+    public MoveMovableSubsystem(MovableSubsystem subsystem, DoubleSupplier power, BooleanSupplier isFinished) {
+        this.subsystem = subsystem;
+        this.power = power;
+        this.isFinished = isFinished;
+        addRequirements(subsystem);
```
no shut the fuck up

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+    }
+
+    public void bindOverrideCommands() {
+        overrideXboxController.getButtonA().whenPressed(subsystemContainer.INTAKE_OPENER_SS::toggleSolenoid);
+        overrideXboxController.getLeftStickButton().whileHeld(new OverrideCommand(subsystemContainer.LOADER_SS, () -> overrideXboxController.getY(Hand.kLeft)));
+        overrideXboxController.getRightStickButton().whileHeld(new OverrideCommand(subsystemContainer.SPINNER_SS, () -> overrideXboxController.getX(Hand.kRight)));
+        overrideXboxController.getButtonX().whenHeld(new ShooterCMD(subsystemContainer.SHOOTER_SS, null, robotConstants.shooterConstants, () -> 3200));
+        overrideXboxController.getButtonB().whileHeld(new ShootWithoutLimelight(subsystemContainer, robotConstants, () -> 3500)).whenReleased(new InstantCommand(() -> {subsystemContainer.PITCHER_SS.setSolenoidState(false);}));
+        overrideXboxController.getButtonY().whileHeld(new ShootCMDGP(subsystemContainer, robotConstants,limelight,  3500));
```
explain in comments

Mboesn committed:
changes in robotContainer and SensorTest

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+
+    public void setShuffleBoard() {
+        SmartDashboard.putNumber("D-Pad", driverXboxController.getPOV());
+        SmartDashboard.putData(" collect ", collectCMDGP);
```
trim

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+        SmartDashboard.putNumber("D-Pad", driverXboxController.getPOV());
+        SmartDashboard.putData(" collect ", collectCMDGP);
+        SmartDashboard.putData("ToggleMotorsModeCMD", toggleMotorsModeCMD);
+        SmartDashboard.putData(" shoot ", ShootWithPitcherCMDGP);
```
trim

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+    }
+
+    public void setShuffleBoard() {
+        SmartDashboard.putNumber("D-Pad", driverXboxController.getPOV());
+        SmartDashboard.putData(" collect ", collectCMDGP);
+        SmartDashboard.putData("ToggleMotorsModeCMD", toggleMotorsModeCMD);
+        SmartDashboard.putData(" shoot ", ShootWithPitcherCMDGP);
+        SmartDashboard.putData("toggle solenoid",
+                new InstantCommand(subsystemContainer.PITCHER_SS::toggleSolenoid, subsystemContainer.PITCHER_SS));
+        subsystemContainer.DRIVETRAIN_SS.setDefaultCommand(supplierFieldDriveCMD);
+        SmartDashboard.putData("Loader/Load", new LoaderCMD(subsystemContainer.LOADER_SS,
+                robotConstants.loaderConstants, robotConstants.loaderConstants.DEFAULT_SHOOTING_VELOCITY));
+        SmartDashboard.putData("Shoot without pitcher CMDGP", shootCMDGP);
+        subsystemContainer.DRIVETRAIN_SS.setDefaultCommand(supplierFieldDriveCMD);
+        SmartDashboard.putData("Pitcher/Open", new InstantCommand(() -> subsystemContainer.PITCHER_SS.setSolenoidState(true)));
+        SmartDashboard.putData("Pitcher/Close", new InstantCommand(() -> subsystemContainer.PITCHER_SS.setSolenoidState(false)));
+        SmartDashboard.putBoolean("Pitcher/State", subsystemContainer.PITCHER_SS.getSolenoidState());
+        SmartDashboard.putData("Drivetrain/ResetDirection", resetDirection);
```
Name them consistantly

laviRZ commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+     */
+    public void autonomousInit() {
+        CommandScheduler.getInstance().schedule(intakeCMD);
+        CommandScheduler.getInstance().schedule(new BackupAuto(subsystemContainer, robotConstants, limelight));
```
explain in comment

laviRZ commented on src/main/java/frc/robot/commands/GenericCalibrateKF.java:
```diff
@@ -0,0 +1,90 @@
+package frc.robot.commands;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.subsystems.KFCallebratableSubsystem;
```
should not

Mboesn replied:
> should not

but like why?

laviRZ commented on src/main/java/frc/robot/commands/GenericCalibrateKF.java:
```diff
@@ -0,0 +1,90 @@
+
+    @Override
+    public void execute() {
+        if (postTest) {
+            subsystem.stopMoving();
+            output += constants.accelerationPerTest;
+            velocitySum = 0;
+            sampleCount = 0;
+            postTest = false;
+        } else {
+            subsystem.move(output);
+            if (atSetpoint()) {
+                velocitySum += subsystem.getVelocity();
+                sampleCount++;
+
+            } else {
+                velocitySum = 0;
+                sampleCount = 0;
+            }
+            if (sampleCount == constants.sampleAmount) {
```
comment and explain the logic

laviRZ commented on src/main/java/frc/robot/commands/GenericCalibrateKF.java:
```diff
@@ -0,0 +1,90 @@
+    @Override
+    public void end(boolean interrupted) {
+        subsystem.stopMoving();
+        System.out.println(KFSum / testCount);
```
don't

laviRZ commented on src/main/java/frc/robot/commands/GenericCalibrateKF.java:
```diff
@@ -0,0 +1,90 @@
+    }
+
+    public boolean atSetpoint() {
+        return Math.abs(subsystem.getVelocity() - lastVelocity) / CODE_ITERATION_RATE < constants.accelerationTolerance;
```
explain in comment

laviRZ commented on src/main/java/frc/robot/motion_profiling/TrigonSwerveControllerCMDGP.java:
```diff
@@ -0,0 +1,59 @@
+        SmartDashboard.putData("TrigonSwerveControllerCMDGP/PID X", constants.X_PID_CONTROLLER);
+        SmartDashboard.putData("TrigonSwerveControllerCMDGP/PID Y", constants.Y_PID_CONTROLLER);
+        SmartDashboard.putData("TrigonSwerveControllerCMDGP/PID Rotation", constants.THETA_PROFILED_PID_CONTROLLER);
+        addCommands(
+                new InstantCommand(() ->
+                {
+                    drivetrainSS.stopMoving();
+                    drivetrainSS.SetSpeedMotorRampRates(0);
+                    drivetrainSS.resetOdometry(
+                            path.getPath(drivetrainSS, constants).getTrajectory().getInitialPose()
+                    );
+                }, drivetrainSS),
+                new SwerveControllerCommand(
+                        path.getPath(drivetrainSS, constants).getTrajectory(),
+                        drivetrainSS::getPose,
+                        drivetrainSS.getKinematics(),
+                        constants.X_PID_CONTROLLER,
+                        constants.Y_PID_CONTROLLER,
+                        constants.THETA_PROFILED_PID_CONTROLLER,
+                        drivetrainSS::setDesiredStates,
+                        drivetrainSS
+                ),
+                new InstantCommand(() -> {
+                    drivetrainSS.SetSpeedMotorRampRates();
+                    drivetrainSS.stopMoving();
+                }, drivetrainSS)
+        );
```
explain in comments

laviRZ commented on src/main/java/frc/robot/motion_profiling/Waypoint.java:
```diff
@@ -0,0 +1,14 @@
+    public Waypoint(double x, double y, double angle) {
+        super(x, y, Rotation2d.fromDegrees(angle));
+    }
+}
```
fu

laviRZ commented on src/main/java/frc/robot/subsystems/led/LedSS.java:
```diff
@@ -0,0 +1,132 @@
+    public void notifierPeriodic() {
+        if (blinkingAmount == 0)
+            setColor(lastColorBeforeBlink);
+        if (blinkingAmount > 0) {
+            LedColor colorToSet;
+            if (blinkingAmount % 2 == 1)
+                colorToSet = LedColor.Off;
+            else
+                colorToSet = blinkColor;
+            setControllerPower(colorToSet.getValue());
+            currentColor = colorToSet;
+            blinkingAmount--;
```
explain in comment

laviRZ commented on src/main/java/frc/robot/subsystems/led/LedSS.java:
```diff
@@ -0,0 +1,132 @@
+     * @return random number between -0.05 to -0.99 in jumps of 0.02
+     */
+    private double getRandomPattern() {
+        double rand = 0.1 * this.rand.nextInt(10);
+        int odd = randomOddNumber();
+        while (odd < 5 && rand == 0.0) {
+            odd = randomOddNumber();
+        }
+        return -(rand + odd * 0.01);
```
eic

laviRZ commented on src/main/java/frc/robot/subsystems/led/LedSS.java:
```diff
@@ -0,0 +1,132 @@
+     */
+    private int randomOddNumber() {
+        int rand = this.rand.nextInt(10);
+        return rand % 2 == 0 ? rand + 1 : rand;
```
eic

laviRZ commented on src/main/java/frc/robot/subsystems/shooter/ShooterCMD.java:
```diff
@@ -0,0 +1,193 @@
+    @Override
+    public void initialize() {
+        shooterSS.setRampRate(constants.SHOOTING_RAMP_RATE);
+        hasRecalculatedF = false;
+        hasSetpoint = false;
+        ballsShotCount = 0;
+        sampleCount = 0;
+        outputSum = 0;
+        lastTimeAtSetpoint = Timer.getFPGATimestamp();
+        currentState = ShooterState.AfterShot;
+        TBHController.reset();
+        PIDController.reset();
+        if (isUsingLimelight)
+            limelight.startVision(Target.PowerPort);
+        if(desiredVelocitySupplier.getAsDouble() < 2000){
+            desiredVelocity = 3200;
+            DriverStationLogger.logToDS("ShooterCMD/tooLowDesiredVelocity: " + desiredVelocitySupplier.getAsDouble());
+        }
+        else
+            desiredVelocity = desiredVelocitySupplier.getAsDouble();
+        f = constants.KF_COEF_A * desiredVelocity + constants.KF_COEF_B;
```
eic

laviRZ commented on src/main/java/frc/robot/subsystems/spinner/SpinnerSS.java:
```diff
@@ -0,0 +1,96 @@
+public class SpinnerSS extends OverridableSubsystem implements Loggable {
+    private final SpinnerConstants constants;
+    private final TrigonTalonSRX motor;
+//    private final TrigonDoubleSolenoid solenoid;
+//    private final ColorSensorV3 colorSensor;
```
wut

laviRZ commented on src/main/java/frc/robot/utilities/TrigonXboxController.java:
```diff
@@ -0,0 +1,146 @@
+
+public class TrigonXboxController extends XboxController {
+    private static final double kIntermittentRumbleTime = 0.15;
+    private final JoystickButton buttonA;
+    private final JoystickButton buttonB;
+    private final JoystickButton buttonX;
+    private final JoystickButton buttonY;
+    private final JoystickButton leftBumper;
+    private final JoystickButton rightBumper;
+    private final JoystickButton leftStickButton;
+    private final JoystickButton rightStickButton;
+    private final JoystickButton backButton;
+    private final JoystickButton startButton;
```
whywhywhywhywhywhwyhwhyw
so ugly

Mboesn committed:
more changes  in krombein review

Mboesn commented on src/main/java/frc/robot/subsystems/led/LedSS.java:
```diff
@@ -0,0 +1,132 @@
+    public LedSS(RobotConstants.LedConstants ledConstants) {
+        ledController = new Spark(ledConstants.PWM_MAP.LED_CONTROLLER);
+        currentColor = LedColor.Off;
+        blinkingAmount = -1;
```
i dont know how the logic works, why dont you try understanding it?
\

Mboesn commented on src/main/java/frc/robot/subsystems/shooter/CalibrateShooterKfCMD.java:
```diff
@@ -0,0 +1,105 @@
+
+        this.shooterSS = shooterSS;
+        this.constants = constants;
+        this.tbhController = new TBHController(constants.TBH_CONTROLLER);
```
very nice

Mboesn committed:
changes according to natan review

switched ledcontroller and blink time to constants
switched isopen place in constructor

Mboesn committed:
change according to natan review that was forgoten

Mboesn committed:
fixed ohads work on the code review

Mboesn committed:
Merge branch 'BranchForCodeReview1' of https://github.com/Programming-TRIGON/RobotCode2021 into BranchForCodeReview1

Mboesn committed:
added DashBoardDataContainer and CommandContrainer

Based on Ohad code review done by me not Ohad

Mboesn commented on src/main/java/frc/robot/RobotContainer.java:
```diff
@@ -0,0 +1,240 @@
+
+    private ShootCMDGP shootCMDGP;
+    private ShootWithPitcherCMDGP ShootWithPitcherCMDGP;
+    private CollectCMDGP collectCMDGP;
```
this comment was brought to you by Ohad not me

Mboesn commented on src/main/java/frc/robot/commands/CommandWithoutSubsystemRequirement.java:
```diff
@@ -0,0 +1,32 @@
+package frc.robot.commands;
```
*do not keep it

Mboesn committed:
ToggleMotorsMode updated

based on ohads code review fixed by me not him

Mboesn committed:
made robot nicer

Mboesn commented on src/main/java/frc/robot/commands/MoveMovableSubsystem.java:
```diff
@@ -0,0 +1,47 @@
+    public boolean isFinished() {
+        return isFinished.getAsBoolean();
+    }
+}
```
Ohad is this sarcasm?

Mboesn commented on src/main/java/frc/robot/commands/MoveMovableSubsystem.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.function.BooleanSupplier;
+import java.util.function.DoubleSupplier;
+
+public class MoveMovableSubsystem extends CommandBase {
```
to what, Ohad? - actual Mboesn (I'm totally not talking to myself)

Mboesn commented on src/main/java/frc/robot/commands/TurnAndPositionToTargetCMD.java:
```diff
@@ -0,0 +1,88 @@
+    private Target target;
+    private TrigonPIDController rotationPIDController;
+    private TrigonPIDController positionPIDController;
+    private double lastTimeSeenTarget, yTarget;
```
ySetpoint it wasn't used because the dashboard value was used instead

Mboesn committed:
continued to fix based on Ohad's code review

Mboesn committed:
set encoder offset and changed get angle in drivetrain

Mboesn committed:
calibrated spinner and loader values and added more logic

Mboesn committed:
Calibrated values and added swerve module calibration as comment

Mboesn committed:
Created Climber SS and CMD (#53)

Co-Authored-By: lycaon7 <65887925+lycaon7@users.noreply.github.com>

Co-authored-by: lycaon7 <65887925+lycaon7@users.noreply.github.com>

Mboesn committed:
fixed issues with swerve

this took way too long but it works hopefully

Mboesn committed:
changed swerve angle profiled pid to normal pid and change values

Mboesn committed:
fixed encoders and allowed driver to shoot without limelight
